### PR TITLE
fix: `GET` requests with `query` parameter if GraphiQL is enabled

### PR DESCRIPTION
**Pull request by schwma**

### CODE REVIEW

File: test/tests/graphiql.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const path = require('path')
+  const { gql } = require('../util')
+
+  const { axios, GET } = cds.test(path.join(__dirname, '../resources/bookshop-graphql'))
+  // Prevent axios from throwing errors for non 2xx status codes
+  axios.defaults.validateStatus = false
+
+  test('GET request to endpoint should serve HTML containing GraphiQL', async () => {
+    const response = await GET('/graphql')
+    expect(response.data).toMatch(/<html>[\s\S]*graphiql[\s\S]*<\/html>/)
```
Comment (ralfhandl) [inline test/tests/graphiql.test.js:12]: Isn't that a lengthy synonym for
```suggestion
    expect(response.data).toMatch(/<html>.*graphiql.*<\/html>/)
```

Comment (schwma) [reply test/tests/graphiql.test.js:12]: No, I tried that at first, but unfortunately `.` doesn't match newline characters in JavaScript

Comment (ralfhandl) [reply test/tests/graphiql.test.js:12]: Learned something today 😄

File: app/graphiql.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const path = require('path')
+
+module.exports = (req, res, next) => {
+  if (req.query.query) {
```
Comment (johannes-vogel) [inline app/graphiql.js:4]: Is the query parameter sufficient or should we also check the accept header?

Comment (ralfhandl) [reply app/graphiql.js:4]: Query parameter should be sufficient. Checking the Accept header risks not being able to test this from the browser, depending on how strict the check is.

Comment (johannes-vogel) [reply app/graphiql.js:4]: fair point. I'm convinced :)

**Review by johannes-vogel — APPROVED** [review]

File: app/graphiql.js (hunk 2 above)
Comment (schwma) [reply app/graphiql.js:4]: I believe GraphiQL 1 had a permalink feature where the `query` parameter would be used to prefill the query textfield in the UI. However, as far as I can tell, this feature does not seem to exist in GraphiQL 2, so there currently does not seem to be a need to redirect some requests to GraphiQL.

**Pull request merged**